### PR TITLE
User can create time record

### PR DESCRIPTION
**Pull request by NiciP**
#### Description
* Components added: Savetimerecording, Timeinputform 
* Module added: Savetimesheet
* API response testing
* Date/time formatting
* Success message displayed
* Feature testing

#### What I have learned working on this feature: 
Sending and fetching API response

#### Screenshots:
image

### CODE REVIEW

File: cypress/integration/userCanCreateTimeRecording.feature.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+describe("User can create time recording", () => {
```
Comment (oliverochman) [inline cypress/integration/userCanCreateTimeRecording.feature.js:3]: add this on top of the file:
`<reference types="Cypress" />`

**Review by yanqiuzhang — APPROVED** [review]

File: cypress/integration/userCanCreateTimeRecording.feature.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+				"X-AUTH-USER": "susan_super",
+				"X-AUTH-TOKEN": "api_kitten"
+			}
+		})
+		cy.visit("http://localhost:3000");
+		cy.get('input[name="begin"]').type("07:00");
+		cy.get('input[name="end"]').type("08:00");
+		cy.get('button[name="create"]').click();
+		cy.contains('Your time was saved')
+	});
```
Comment (oliverochman) [inline cypress/integration/userCanCreateTimeRecording.feature.js:39]: How much have you restricted the input fields? You should add a sad pat here

File: src/Modules/SaveTimesheet.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+import moment from 'moment';
+
+const saveTime = async (begin, end) => {
+	const apiUrl = "https://demo-stable.kimai.org/api/";
+	const proxyUrl = "https://cors-anywhere.herokuapp.com/";
```
Comment (oliverochman) [inline src/Modules/SaveTimesheet.js]: In other features I have worked on, we dont need to use the proxyUrl. Can you try to remove it again and see if it still works?

File: src/components/SaveTimeRecording.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+class SaveTimeRecording extends Component {
+	state = {
+		begin: "",
+		end: "",
+		saveTime: false
+	};
+
+	async postTimesheets() {
+		let response = await saveTime(this.state.begin, this.state.end);
```
Comment (oliverochman) [inline src/components/SaveTimeRecording.js]: are you guys using the response variable for something?

File: src/components/SaveTimeRecording.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React, { Component } from "react";
+import { saveTime } from "../Modules/SaveTimesheet";
+import TimeInputForm from "./TimeInputForm";
+import { Form, Button, Grid } from "semantic-ui-react";
+
+class SaveTimeRecording extends Component {
+	state = {
+		begin: "",
+		end: "",
+		saveTime: false
```
Comment (oliverochman) [inline src/components/SaveTimeRecording.js]: I would change this to something more descriptive like `timeSaved`. That would make more sense due to how you use that state.

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/Modules/SaveTimesheet.js (hunk 3 above)
Comment (NiciP) [reply src/Modules/SaveTimesheet.js]: Error 403: Forbidden and Network error received when doing so

Comment (Joberg95) [reply src/Modules/SaveTimesheet.js]: i also think we need it since making post requests from localhost seems to a big "nono"

**Review by yanqiuzhang — APPROVED** [review]

File: cypress/integration/userCanCreateTimeRecording.feature.js (hunk 2 above)
Comment (Joberg95) [reply cypress/integration/userCanCreateTimeRecording.feature.js:39]: Added

File: cypress/integration/userCanCreateTimeRecording.feature.js (hunk 1 above)
Comment (Joberg95) [reply cypress/integration/userCanCreateTimeRecording.feature.js:3]: added

Comment (NiciP) [reply cypress/integration/userCanCreateTimeRecording.feature.js:3]: added

**Pull request merged**